core/imt/src/RArena.hxx

```diff
@@ -0,0 +1,20 @@
+#include "tbb/task_arena.h"
+namespace ROOT {
+namespace Internal {
+    // Wrapper for tbb::task_arena.
+    //
+    //tbb::task_arena is an alias of tbb::interface7::task_arena, which doesn't allow
+    // to forward declare tbb::task_arena. To avoid code breaking on tbb interface changes
+    // we don't forward declare tbb::interface7::task_arena and instead we wrap tbb::task_arena
+    // in the forward declared class RArena
+    class RArenaPtr {
+    public:
+    /// Access the wrapped object and allow to call its methods.
+    tbb::task_arena *operator->() {
+        return fTBBArena.get();
```

Comment:
now that I think about it...doesn't this completely break ROOT's encapsulation/hiding of TBB? client code can call any task_arena method via TPoolManager (and it does, see all the calls to execute, terminate, etc.). Is this what we want? Doesn't it make it harder to switch the backend to e.g. hpx?

Reply:
This headers are not exposed (/imt/src), and the class is ROOT::Internal anyway... wouldn't that be enough?

Reply:
In any case, there's the proposal in the last question of the description. We'll discuss it in the PPP meeting, but that should remove the public call to Arena()

Reply:
i just realized i'm not clear on our encapsulation/hiding requirements for TBB :sweat_smile:

independently of that, isn't it the case that the more TBB internals we expose to ROOT's MT machinery, the harder it becomes to swap TBB for something else?